pkg/kubelet/winstats/version.go

```diff
@@ -0,0 +1,95 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package winstats
+
+import (
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// getVersionRevision gets revision from UBR registry.
+func getVersionRevision() (uint16, error) {
+	var h windows.Handle
+	if err := windows.RegOpenKeyEx(windows.HKEY_LOCAL_MACHINE,
+		windows.StringToUTF16Ptr(`SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\`),
+		0,
+		windows.KEY_READ,
+		&h); err != nil {
+		return 0, err
+	}
+	defer windows.RegCloseKey(h)
+
+	var buf [1 << 10]uint16
```

Comment:
nit: does buf need to be that large? What are typical values of this key?

Reply:
It's the value of windows registy, may includes integers, uint arrays or string.

Reply:
Generally, yes registry values could be any of those. But in this specific case, you must have some expectations for what the 2 fields you're querying can hold? E.g. if it's a version string it's probably not more than, what, 32 characters?  Allocating 2KB everytime this is called seems like kind of a lot.

Reply:
Good catch. I see all other places using this API are allocating such size, but it is not required such big in this package.